src/widget.jl

```diff
@@ -7,26 +7,28 @@ mutable struct Widget{T, S} <: AbstractWidget{T, S}
     output::AbstractObservable{S}
     scope
     layout::Function
-    function Widget{T}(components::OrderedDict{Symbol, Any};
-        output = Observable{Any}(nothing),
-        scope = nothing,
-        layout = defaultlayout(get_backend())) where {T, S}
-
-        output_obs = to_abstractobservable(output)
-        new{T, eltype(output_obs)}(components, to_abstractobservable(output), scope, layout)
+    function Widget{T}(components::OrderedDict{Symbol,Any}, output::Observable{S}, scope, layout) where {T,S}
+        new{T,S}(components, output, scope, layout)
     end
 end
 
+function Widget{T}(components::OrderedDict{Symbol,Any};
+    output = Observable{Any}(nothing),
+    scope = nothing,
+    layout = defaultlayout(get_backend())
+) where {T}
+    output_obs = to_abstractobservable(output)
```

Comment:
Actually, instead of `to_abstractobservable` (which I suspect one could remove), maybe the following is a bit better:

```
output_obs = output isa AbstractObservable ? observe(output) : Observable(output)
```

as `observe` tends to transform abstract observables into actual `Observable` objects, so would give us more compatibility (in case the output is itself a widget for example, which is somewhat unlikely but possible).